### PR TITLE
[FIX] Infinite calls to updated api

### PR DESCRIPTION
**Pull request by rafaelbiten**
**Change 1.** We recently started seeing jenkins failing to run with the following error:

```
TestRail API returned HTTP 429 ({'error': 'API Rate Limit Exceeded - 180 per minute maximum allowed. Retry after 1 seconds.'})
```

After investigating the issue, it seems like the api behind the calls to `self.send_get(f'get_cases...')` have changed and no longer fails (or returns empty) when we no longer have items to receive.

We changed the code so we only make that call a maximum of 10 times and we also stop making extra calls when we stop receiving the expected cases.

**Change 2.** The code was also failing (for an `unknown to me` reason) in comprehensions like `[plan['name'] for plan in plans]` with `TypeError: string indices must be integers`, so the code was changed in a couple of other places too.

### CODE REVIEW

**Review by dbernar — APPROVED** [review]
👍

File: utils/testrail/testrail_service.py [hunk 1]
```diff
@@ -110,14 +110,28 @@ def get_attachments_for_test(self, test_id):
     def get_case(self, case_id: str):
         return self.send_get(f'get_case/{case_id}')
 
+    # TODO: get_cases and is_response_with_more_cases below should rely on the api
```
Comment (fiska-rafal) [inline utils/testrail/testrail_service.py:113]: `TODO` needs a name for the poor unfortunate soul that may need to work on this.

**Review by fiska-rafal — APPROVED** [review]
👍

**Review by dbernar — APPROVED** [review]
👍

File: utils/testrail/testrail_service.py (hunk 1 above)
Comment (rafaelbiten) [reply utils/testrail/testrail_service.py]: Added my name in there...

**Comment by rafaelbiten** [comment]
I'll merge this PR to see if we can run the tests against our master branch successfully. Will leave the branch in place, in case we need anything else.

**Comment by rafaelbiten** [comment]
The tests are running as expected in Jenkins and we're back to stable results, so we're deleting this PR.

**Pull request merged**